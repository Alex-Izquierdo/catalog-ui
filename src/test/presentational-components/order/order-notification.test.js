import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { CLEAR_NOTIFICATIONS } from '@redhat-cloud-services/frontend-components-notifications';

import OrderNotification from '../../../presentational-components/order/order-notification';

describe('<OrderNotification />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      id: '123',
      dispatch: jest.fn()
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <MemoryRouter>
        <OrderNotification {...initialProps} />
      </MemoryRouter>
    );
    expect(shallowToJson(wrapper.find(OrderNotification))).toMatchSnapshot();
  });

  it('should call dispatch on link click with correct arguments', () => {
    const dispatch = jest.fn();
    const wrapper = mount(
      <MemoryRouter>
        <OrderNotification {...initialProps} dispatch={dispatch} />
      </MemoryRouter>
    );
    wrapper.find('a').simulate('click');
    expect(dispatch).toHaveBeenCalledWith({ type: CLEAR_NOTIFICATIONS });
  });
});
