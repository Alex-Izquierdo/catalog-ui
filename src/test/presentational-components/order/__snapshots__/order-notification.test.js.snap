// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderNotification /> should render correctly 1`] = `
<OrderNotification
  dispatch={[MockFunction]}
  id="123"
/>
`;
